pkg/cloudprovider/azure.go

```diff
@@ -52,9 +143,218 @@ func (p *AzureProvider) Validate(secret *kapi.Secret) []string {
 }
 
 func (p *AzureProvider) Test(secret *kapi.Secret) error {
+	var err error
+
 	if secret == nil {
 		return nil
 	}
 
+	switch p.Role {
+	case BackupStorage:
+		cloudCreds, err := godotenv.Unmarshal(string(secret.Data[AzureCredentials]))
+		if err != nil {
+			return err
+		}
+
+		blobClient, err := p.getBlobClient(cloudCreds)
+		if err != nil {
+			return err
+		}
+
+		key, _ := uuid.NewUUID()
+		test := AzureBlobTest{
+			key:       key.String(),
+			container: p.StorageContainer,
+			client:    blobClient,
+		}
+		err = test.Run()
+	}
+
+	return err
+}
+
+func (p *AzureProvider) getBlobClient(cloudCreds map[string]string) (*azstorage.BlobStorageClient, error) {
```

Comment:
This is only used by the `AzureBlobTest`.  Seems simpler for the `AzureBlobTest` to have a field for the _secret_ and `getBlobClient()` be a method.

Reply:
This seems like a good change, will implement.

Reply:
Changed.